types/wlr_input_method_v2.c

```diff
@@ -37,6 +55,9 @@ static void input_method_resource_destroy(struct wl_resource *resource) {
 	if (!input_method) {
 		return;
 	}
+	if (input_method->keyboard_grab != NULL) {
+		wl_resource_destroy(input_method->keyboard_grab->resource);
```

Comment:
We can't destroy resources when the client hasn't requested it. We need to make the resource inert instead. See CONTRIBUTING.

Reply:
Making the resource inert is already done in `keyboard_grab_destroy` (called in `input_method_destroy`). So we should be able to just remove this `wl_resource_destroy` call.

Reply:
I've removed it but I found another problem. The protocol specified that `destroy` request on zwp_input_method_v2 should also destroy associated child like zwp_input_method_keyboard_grab_v2. If input method is already inert, how can this be done? How can I find resource of grab to call wl_resource_destroy? Should I add a layer to input method user data to track child object resource?

Reply:
Hmm, that is troublesome. I think right before the `wl_resource_destroy` call for the zwp_input_method_v2 resource we'll want to also call `wl_resource_destroy` on the zwp_input_method_keyboard_grab_v2 resource, if any.

Reply:
If input method is inert, struct `wlr_input_method_v2` is already freed. We need another way to find wl_resource of the grab.

Reply:
If the input method is inert, it means we've already free'ed the `wlr_input_method_v2` (and the grab with it). So it should be fine.